### PR TITLE
Fix #111: Add support for new framework balance APIs

### LINKED ISSUE #111: Support new balances API in UI
After #102 is merged [in core](https://github.com/brave/brave-core/pull/2825)

The app currently uses `BraveLedger.balance` and `BraveLedger.walletInfo.rates` extensively which have now been moved into a new `Balance` struct located at `BraveLedger.balance`

### PR DESCRIPTION
**Pull request by kylehickinson**
Some notes:
- CoreData model has been updated in the BraveRewards.framework but didn't use a new version in this case so any crashes to due with CoreData may require you to just delete app and reinstall
- `build_in_core.sh` script now requires a clean git directory in both `brave-browser` and `brave-browser/src/brave` if you don't specify the `--skip-update` flag (to avoid clobbering your changes if you happen to forget the flag)
- Error text when creating a wallet needs to be fixed at a future date (will open a ticket)

### CODE REVIEW

File: BraveRewardsUI/Settings/Tips/Details/TipsDetailViewController.swift [hunk 1]
```diff
@@ -126,7 +126,8 @@ extension TipsDetailViewController: UITableViewDataSource, UITableViewDelegate {
     guard let typedSection = Section(rawValue: section) else { return 0 }
     switch typedSection {
     case .summary:
-      return 1 + (tipsList.filter({$0.category == .recurringTip}).isEmpty ? 0 : 1 )
+      let recurringTips = tipsList.filter({$0.rewardsCategory == .recurringTip})
```
Comment (iccub) [inline BraveRewardsUI/Settings/Tips/Details/TipsDetailViewController.swift]: nitpick, maybe use
`tipsList.compactMap { $0.rewardsCategory }.contains(.recurringTip)` or
`tipsList.contains { $0.rewardsCategory == .recurringTip } `

File: lib/BraveRewards.framework/Headers/BATBraveLedger.h [hunk 2]
```diff
@@ -48,17 +48,24 @@ NS_SWIFT_NAME(BraveLedger)
...
+/// The users current wallet balance and related info
+@property (nonatomic, readonly, nullable) BATBalance *balance;
+
 /// The wallet's passphrase. nil if the wallet has not been created yet
 @property (nonatomic, readonly, nullable) NSString *walletPassphrase;
 
 /// Recover the users wallet using their passphrase
-- (void)recoverWalletUsingPassphrase:(NSString *)passphrase completion:(nullable void (^)(NSError *_Nullable))completion;
+- (void)recoverWalletUsingPassphrase:(NSString *)passphrase
+                          completion:(nullable void (^)(NSError *_Nullable))completion;
```
Comment (iccub) [inline lib/BraveRewards.framework/Headers/BATBraveLedger.h]: add space between `*` and `_Nullable`

File: BraveRewardsUI/Extensions/BraveLedgerExtensions.swift [hunk 3]
```diff
@@ -74,4 +80,51 @@ extension BraveLedger {
...
+      }
+      group.enter()
+      self.fetchBalance { balance in
+        if balance == nil {
+          success = false
+        }
+        group.leave()
+      }
+      group.notify(queue: .main, execute: {
+        completion(success)
```
[the remark this answers is not in the document]
Comment (kylehickinson) [reply BraveRewardsUI/Extensions/BraveLedgerExtensions.swift]: Even if either of the calls fail `notify` should still be called anyways as long as the completion blocks are executed and therefore calling `group.leave()` :)

File: lib/BraveRewards.framework/Headers/BATBraveLedger.h (hunk 2 above)
Comment (kylehickinson) [reply lib/BraveRewards.framework/Headers/BATBraveLedger.h]: This will have to be done over in https://github.com/brave/brave-core/pull/2832

**Pull request merged**